### PR TITLE
701 processing mc flag

### PR DESCRIPTION
**Pull request by garyt2**
Adds a usable flag to process input.py instructions independent of the enabled flag

### CODE REVIEW

File: include/trick/MonteCarlo.hh [hunk 1]
```diff
@@ -449,6 +454,20 @@ namespace Trick {
          */
         bool get_enabled();
 
+        /**
+         * Sets #processing_input_for_MC.
+         *
+         * @see @ref MonteCarloEnabling "Enabling Monte Carlo"
```
Comment (garyt2) [inline include/trick/MonteCarlo.hh]: This has not been updated; need guidance from Trick team on how best to do this.

File: include/trick/montecarlo_c_intf.h [hunk 2]
```diff
@@ -23,6 +23,18 @@ void mc_set_enabled(int enabled);
  */
 int mc_get_enabled(void);
 
+/**
+ * @relates Trick::MonteCarlo
+ * @copydoc MonteCarlo::set_processing_input_for_MC
```
Comment (garyt2) [inline include/trick/montecarlo_c_intf.h]: I'm not sure what this does, may need to add something to documentation before merging or change this line.

File: trick_source/sim_services/MonteCarlo/MonteCarlo_funcs.cpp [hunk 3]
```diff
@@ -11,12 +11,23 @@
 
 void Trick::MonteCarlo::set_enabled(bool in_enabled) {
     this->enabled = in_enabled;
+    if (in_enabled) {
+      set_processing_input_for_MC( true);
```
Comment (shahzadlone) [inline trick_source/sim_services/MonteCarlo/MonteCarlo_funcs.cpp]: Fix space.
```diff
-      set_processing_input_for_MC( true);
+      set_processing_input_for_MC(true);
```

**Comment by alexlin0** [comment]
I believe that we have the ability to do this without code modification.  I believe that you can test if you are a slave and can test if monte_carlo is enabled, you achieve the desired effect.

```python
if trick.mc_is_slave() and trick.mc_get_enabled():
    print "I am a slave processing input file"
```

**Comment by garyt2** [comment]
I think this has become OBE, so I'll close it.  The accumulated implications of trying to piece together patches to cover the use cases we need made doing so unworkable ... suggesting instead the need for an independent refactor/ comprehensive interface which is now underway.

The driver for this work was to get a mechanism in place to allow input file processing during re-runs to match that during first-runs.  Because `mc_get_enabled()` returns false during a re-run, any logic embedded in `if trick.mc_get_enabled()` will not be processed during a re-run, the slave status is moot.  Consequently the processing of the input files will still not match so no, this does not appear to provide the solution.

**Pull request closed without merging**